src/status_im/ui2/screens/chat/components/chat_item.cljs

```diff
@@ -0,0 +1,199 @@
+                 :accessibility-label :chat-message-text}
+        (:text content)]
+       [render-subheader (:parsed-text content)])
+
+     (= constants/content-type-sticker content-type)
+     [preview-label :t/sticker]
+
+     (= constants/content-type-image content-type)
+     [preview-label :t/image]
+
+     (= constants/content-type-audio content-type)
+     [preview-label :t/audio]
+
+     (content-type-community-invite? content-type community-id)
+     (let [{:keys [name]}
+           @(re-frame/subscribe [:communities/community community-id])]
+       [preview-label :t/community-message-preview {:community-name name}]))])
+
+(def memo-timestamp
+  (memoize
+   (fn [timestamp]
+     (string/upper-case (time/to-short-str timestamp)))))
+
+(defn unviewed-indicator [{:keys [unviewed-mentions-count
+                                  unviewed-messages-count
+                                  public?]}]
+  (when (pos? unviewed-messages-count)
+    [rn/view {:position :absolute :right 16}
+     (cond
+       (and public? (not (pos? unviewed-mentions-count)))
+       [rn/view {:style                  styles/public-unread
+                 :accessibility-label :unviewed-messages-public}]
+
+       (and public? (pos? unviewed-mentions-count))
+       [badge/message-counter unviewed-mentions-count]
+
+       :else
+       [badge/message-counter unviewed-messages-count])]))
+
+(defn icon-style []
+  {:color           colors/black
+   :width           15
+   :height          15
+   :container-style {:top          13 :left 72
+                     :position     :absolute
+                     :width        15
+                     :height       15
+                     :margin-right 2}})
+
+(defn chat-item-icon [muted private-group? public-group?]
+  (cond
+    muted
+    [icons/icon :main-icons/tiny-muted (assoc (icon-style) :color colors/gray)]
+    private-group?
+    [icons/icon :main-icons/tiny-group (icon-style)]
+    public-group?
+    [icons/icon :main-icons/tiny-public (icon-style)]
+    :else
+    [icons/icon :main-icons/tiny-new-contact (icon-style)]))
+
+(defn chat-item-title [chat-id muted group-chat chat-name edit?]
+  [quo2.text/text {:weight              :semi-bold
+                   :color               (when muted :secondary)
+                   :accessibility-label :chat-name-text
+                   :ellipsize-mode      :tail
+                   :number-of-lines     1
+                   :style               {:position :absolute
+                                         :left     72
+                                         :top      10
+                                         :right    (if edit? 50 90)}}
+   (if group-chat
+     (utils/truncate-str chat-name 30)
+     ;; This looks a bit odd, but I would like only to subscribe
+     ;; if it's a one-to-one. If wrapped in a component styling
+     ;; won't be applied correctly.
+     (first @(re-frame/subscribe [:contacts/contact-two-names-by-identity chat-id])))])
```

Comment:
[Nitpick] I don't find this line odd at all, so to me, at least, I'd prefer the code without this comment.